lightning/src/chain/keysinterface.rs

```diff
@@ -1015,15 +1014,14 @@ impl KeysManager {
 					Err(_) => panic!("Your RNG is busted"),
 				};
 				let channel_master_key = master_key.ckd_priv(&secp_ctx, ChildNumber::from_hardened_idx(3).unwrap()).expect("Your RNG is busted");
-				let rand_bytes_master_key = master_key.ckd_priv(&secp_ctx, ChildNumber::from_hardened_idx(4).unwrap()).expect("Your RNG is busted");
 				let inbound_payment_key: SecretKey = master_key.ckd_priv(&secp_ctx, ChildNumber::from_hardened_idx(5).unwrap()).expect("Your RNG is busted").private_key;
 				let mut inbound_pmt_key_bytes = [0; 32];
 				inbound_pmt_key_bytes.copy_from_slice(&inbound_payment_key[..]);
 
-				let mut rand_bytes_unique_start = Sha256::engine();
-				rand_bytes_unique_start.input(&starting_time_secs.to_be_bytes());
-				rand_bytes_unique_start.input(&starting_time_nanos.to_be_bytes());
-				rand_bytes_unique_start.input(seed);
+				let mut nonce = Vec::new();
+				nonce.append(&mut starting_time_secs.to_be_bytes().to_vec());
+				nonce.append(&mut starting_time_nanos.to_be_bytes().to_vec());
+				let chacha = Mutex::new(ChaCha20::new(seed, &nonce));
```

Comment:
I think we can avoid three vec allocations by having a fixed array and copying the bytes into it.

Reply:
Thanks! I've changed to that approach in [fac7b4e](https://github.com/lightningdevkit/rust-lightning/pull/1974/commits/fac7b4e02777819868f68d539506ef70ed6ed36a).